packages/playground/data-liberation/src/wordpress-core-html-api/class-wp-html-processor.php

```diff
@@ -1613,7 +1613,10 @@ private function step_in_head(): bool {
 				 */
 				$charset = $this->get_attribute( 'charset' );
 				if ( is_string( $charset ) && 'tentative' === $this->state->encoding_confidence ) {
-					$this->bail( 'Cannot yet process META tags with charset to determine encoding.' );
+					// Commenting this out for now. We're assuming UTF-8 in WP_HTML_To_Blocks and
+					// we don't want to fail just because a document contained a meta tag with a UTF-8 charset.
+					// @TODO: Bail on non-utf8 charsets.
+					// $this->bail( 'Cannot yet process META tags with charset to determine encoding.' );
```

Comment:
CC @sirreal – baling out only on non-utf8 charsets would unlock converting full HTML documents that often contain these meta tags.

Reply:
I think the logic is correct here, the HTML processor itself never has problems because the full processor requires UTF-8 and [therefore sets its encoding and encoding confidence to certain](https://github.com/WordPress/wordpress-develop/blob/7af04695d86ebeeb0d3f314c53f1f425a14323a7/src/wp-includes/html-api/class-wp-html-processor.php#L344-L345), in the [fragment processor it's set to irrelevant](https://github.com/WordPress/wordpress-develop/blob/7af04695d86ebeeb0d3f314c53f1f425a14323a7/src/wp-includes/html-api/class-wp-html-processor.php#L547).

I think bailing is correct, everything is required to be utf-8 for correct processing. Create methods should set the encoding_confidence appropriately. Is there are subclassed `create_` method you have that could set the encoding_confidence?